pyocm/pyocm.py

```diff
@@ -5,41 +5,46 @@
 
 class ocm_api:
     """."""
-    def __init__(self, api="https://api.openshift.com/api/", 
+    def __init__(self, api="https://api.openshift.com/api/",
             ca_path="/etc/pki/tls/certs/ca-bundle.crt", token=None):
 
         if token is None:
             raise Exception("Token was not supplied.")
 
+        self.thread_pool = ThreadPoolExecutor(4)
         self.base_api_uri = api
         self.auth_token = self.__confirm_auth(token)
 
         self.ca_certificate_path = ca_path
         if ca_path == False: # Used to Disable SSL warning (from each api call)
             requests.packages.urllib3.disable_warnings()
 
-        self.thread_pool = ThreadPoolExecutor(4)
-
 
     def __del__(self):
         self.thread_pool.shutdown(wait=True)
 
 
     def __confirm_auth(self, token):
-        r = requests.post('https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token', 
+        endpoint = 'https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token'
+        r = requests.post(endpoint,
```

Comment:
This keeps the exception (below, this statement) from throwing an error, due to a missing 'endpoint' variable. It fixes a copy paste error we had.

Reply:
Can we also add a check for `self.thread_pool` existing on line 25?  This should work...

```python
    def __del__(self):
        if self.thread_pool:
            self.thread_pool.shutdown(wait=True)
```